### PR TITLE
feat: support TLS session resumption through session tickets

### PR DESCRIPTION
**Pull request by nettoclaudio**
This PR adds support to resume TLS sessions through session tickets (as described on RFC 5077). It handles the encryption keys rotation under Secret resource and on Nginx runtime as well.

This feature can be activated on RpaasInstance specification under the `tlsSessionResumption` field, e.g.:

```yaml
apiVersion: extensions.tsuru.io/v1alpha1
kind: RpaasInstance
metadata:
  name: example
spec:
  tlsSessionResumption:
      sessionTicket:
        keepLastKeys: 3 # number of keys: 4 = 3 + 1 (last older keys + current one) (defaults to zero)
        keyRotationInterval: 360  # in minutes (defaults to an hour)
        keyLength: 80 # can be either 48 or 80 bytes (defaults to 48 bytes)
```

### CODE REVIEW

File: pkg/controller/rpaasinstance/rpaasinstance_controller.go [hunk 1]
```diff
@@ -354,6 +438,285 @@ func (r *ReconcileRpaasInstance) listDefaultFlavors(ctx context.Context, instanc
...
+		image = instance.Spec.TLSSessionResumption.SessionTicket.Image
+	}
+
+	var secretName string
+	if secret != nil {
+		secretName = secret.Name
+	}
+
+	var secretNamespace string
+	if secret != nil {
```
Comment (wpjunior) [inline pkg/controller/rpaasinstance/rpaasinstance_controller.go:562]: I suppose our never create resource in other odd namespace, why this namespace logic here ?, we could ever assume instance.Namespace ?

File: pkg/controller/rpaasinstance/rpaasinstance_controller.go [hunk 2]
```diff
@@ -763,6 +1126,23 @@ func newNginx(instance *v1alpha1.RpaasInstance, plan *v1alpha1.RpaasPlan, config
...
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretNameForTLSSessionTickets(instance),
+				},
+			},
+		})
+
+		n.Spec.PodTemplate.VolumeMounts = append(n.Spec.PodTemplate.VolumeMounts, corev1.VolumeMount{
+			Name:      sessionTicketsVolumeName,
+			MountPath: sessionTicketsVolumeMountPath,
+			ReadOnly:  true,
```
Comment (wpjunior) [inline pkg/controller/rpaasinstance/rpaasinstance_controller.go:1142]: Nice catch @nettoclaudio

File: pkg/apis/extensions/v1alpha1/rpaasinstance_types.go [hunk 3]
```diff
@@ -79,6 +79,12 @@ type RpaasInstanceSpec struct {
 	// some event happens to nginx container.
 	// +optional
 	Lifecycle *nginxv1alpha1.NginxLifecycle `json:"lifecycle,omitempty"`
+
+	// TLSSessionResumption configures the instance to support session resumption
+	// using either session tickets or session ID (in the future). Defaults to
+	// disabled.
+	// +optional
+	TLSSessionResumption *TLSSessionResumption `json:"tlsSessionResumption,omitempty"`
```
[the remark this answers is not in the document]
Comment (nettoclaudio) [reply pkg/apis/extensions/v1alpha1/rpaasinstance_types.go:87]: They were purposely segregated. Session ticket is not the only method to address TLS session resumption. My idea behind this design is to support session ID in a long long future as well.

[the remark this answers is not in the document]
Comment (wpjunior) [reply pkg/apis/extensions/v1alpha1/rpaasinstance_types.go:87]: To cover this support I agree with you.

File: pkg/controller/rpaasinstance/rpaasinstance_controller.go (hunk 1 above)
Comment (nettoclaudio) [reply pkg/controller/rpaasinstance/rpaasinstance_controller.go:562]: Absolutely.

File: pkg/controller/rpaasinstance/rpaasinstance_controller.go [hunk 4]
```diff
@@ -354,6 +438,285 @@ func (r *ReconcileRpaasInstance) listDefaultFlavors(ctx context.Context, instanc
...
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !enabled {
+		return nil, r.client.Delete(ctx, &secret)
+	}
+
+	newData := make(map[string][]byte)
```
[the remark this answers is not in the document]
Comment (nettoclaudio) [reply pkg/controller/rpaasinstance/rpaasinstance_controller.go:480]: Sure.

**Comment by wpjunior** [comment]
LGTM @nettoclaudio great job!

**Pull request merged**